public/components/register_model/pretrainedmodel_select.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useState, Fragment, useEffect } from 'react';
+import {
+  EuiSpacer,
+  EuiTextColor,
+  EuiSelectable,
+  EuiLink,
+  EuiSelectableOption,
+  EuiHighlight,
+} from '@elastic/eui';
+import { useHistory } from 'react-router-dom';
+import { generatePath } from 'react-router-dom';
+import { modelRepositoryManager } from '../../utils/model_repository_manager';
+import { routerPaths } from '../../../common/router_paths';
+interface IItem {
+  label: string;
+  checked?: 'on' | undefined;
+  description: string;
+}
+interface Props {
+  getPreSelected: (val: boolean) => void;
+}
+const renderModelOption = (option: IItem, searchValue: string) => {
+  return (
+    <>
+      <EuiHighlight search={searchValue}>{option.label}</EuiHighlight>
+      <br />
+      <EuiTextColor color="subdued">
+        <small>
+          <EuiHighlight search={searchValue}>{option.description}</EuiHighlight>
+        </small>
+      </EuiTextColor>
+    </>
+  );
+};
+export const PreTrainedModelSelect = ({ getPreSelected }: Props) => {
+  useEffect(() => {
+    const subscribe = modelRepositoryManager.getPreTrainedModels$().subscribe((models) => {
+      setModelRepoSelection(
+        Object.keys(models).map((name) => ({
+          label: name,
+          description: models[name].description,
+          checked: undefined,
+        }))
+      );
+    });
+    return () => {
+      subscribe.unsubscribe();
+    };
+  }, []);
+  const ShowRest = useCallback(
```

Comment:
This function could be removed.

Reply:
I design this function to control the rest showing after the pretrained model is seletcted. If removed, maybe this function will be failed.